tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot/README.md

```diff
@@ -0,0 +1,85 @@
+﻿# EchoBot using Slack Adapter
+
+Bot Framework v4 echo bot using Slack Adapter sample.
+
+This bot has been created using [Bot Framework](https://dev.botframework.com), it shows how to create a simple echo bot that connects with Slack to respond messages.
+
+## Prerequisites
+
+- [.NET Core SDK](https://dotnet.microsoft.com/download) version 2.1
+
+  ```bash
+  # determine dotnet version
+  dotnet --version
+  ```
+
+## To try this sample
+
+1 - Clone the repository
+
+    ```bash
+    git clone https://github.com/Microsoft/botbuilder-dotnet.git
+    ```
+
+2 - In a terminal, navigate to `test/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot`
+
+3 - Connect the bot with Slack by following the instructions below.
+
+4 - Run the bot from a terminal or from Visual Studio, choose option A or B.
+
+  A) From a terminal
+
+  ```bash
+  # run the bot
+  dotnet run
+  ```
+
+  B) Or from Visual Studio
+
+  - Launch Visual Studio
+  - File -> Open -> Project/Solution
+  - Navigate to `test/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot` folder
+  - Select `Microsoft.Bot.Builder.Adapters.Slack.TestBot.csproj` file
+  - Press <kbd>F5</kbd> to run the project
+
+5 - Using a tunneling tool like [Ngrok](https://ngrok.com/download), expose the bot's endpoint.
+
+6 - Using Slack, in Apps section, search the bot by the _App Name_ and send a message to it.
+
+### Connect the bot with Slack
+
+1 - Create a SlackApp following this [guide](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-connect-slack?view=azure-bot-service-4.0)
+
+Some things to consider:
+
+a. In the **Oauth & Permissions** tab, the _Redirect URL_ should be the one that generated by ngrok.

```

Comment:
The bot must be running with a valid VerificationToken and ClientSigningSecret in order to verify the Redirect Url. These steps are not included in this Readme, and it took a bit for me to figure that out.

Reply:
We'll review this and update the Readme. Thanks!

Reply:
Done. Thanks!